File: java/Ex3_1/BubbleSort.java
public class BubbleSort {
    

    public static int[] bubbleSort(int[] list)
    {

        boolean swapped = false;
        int n = list.length -1 ;
        do{
            swapped = false;

            for(int i = 0 ; i < n;i++)
            {
                if(list[i] > list[i+1])
                {
                    int temp = list[i+1];
                    list[i+1] = list[i];
                    list[i] = temp;
                    swapped = true;

                }
            }
            n--;

        }
        while(swapped);
        return list;
    }
    public static int[] bubbleSortOptimized(int[] list)
    {

        int n = list.length -1 ;
        for(int i = 0 ; i < n;i++)
        {
            for(int j = 0 ; j <n - i;j++)
            {
                if(list[j] > list[j + 1]){
                int temp  = list[j + 1];
                list[j+1] = list[j];
                list[j] = temp;
                }
            }
        }
        return list;
    }
    public static void main(String[] args) {
        
        int[] myList = {2,2,3,1,4,1,3,5};
        int[] sorted = bubbleSortOptimized(myList);
        for(int i : sorted)
        {
            System.out.print(i+" ,");
        }
    }
    

}

File: cpp/sortingProblems/bubbleSort.cpp
#include <iostream>

class BubbleSort
{
public:
    // WRITE YOUR CODE HERE
    static void bubbleSort(int arr[], int n)
    {
        bool swap;
        do
        {
            swap = false;
            for (int i = 0; i < n - 1; i++)
            {
                if (arr[i + 1] < arr[i])
                {
                    int temp = arr[i + 1];
                    arr[i + 1] = arr[i];
                    arr[i] = temp;
                    swap = true;
                }
            }
        } while (swap);
    }
};

int main()
{
    int arr[] = {62, 88, 17, 95, 34};
    int n = sizeof(arr) / sizeof(arr[0]);
    BubbleSort::bubbleSort(arr, n);

    std::cout << "Sorted array:" << std::endl;
    for (int i = 0; i < n; i++)
    {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}
